docs/constrained-generators/manual.md

```diff
@@ -1,28 +1,109 @@
+        - [Function symbols for numeric types](#function-symbols-for-numeric-types)
+        - [` Function symbols for Bool`](#-function-symbols-for-bool)
+        - [Function symbols for List](#function-symbols-for-list)
+        - [Function symbols for Set](#function-symbols-for-set)
+        - [Function symbols for Map](#function-symbols-for-map)
+    - [Generating from and checking against specifications](#generating-from-and-checking-against-specifications)
+    - [How we solve the constraints](#how-we-solve-the-constraints)
+        - [How to pick the variable order](#how-to-pick-the-variable-order)
+        - [The total definition requirement ](#the-total-definition-requirement)
+        - [Using Match to introduce new variables for subcomponents](#using-match-to-introduce-new-variables-for-subcomponents)
+    - [Overloaded types in the library](#overloaded-types-in-the-library)
+- [Library functions to build Term, Pred, Specification](#library-functions-to-build-term-pred-specification)
+    - [From Term to Pred](#from-term-to-pred)
+    - [For all elements in a container type (List, Set, Map)](#for-all-elements-in-a-container-type-list-set-map)
+    - [Reification](#reification)
+    - [Disjunction, choosing between multiple things with the same type](#disjunction-choosing-between-multiple-things-with-the-same-type)
+    - [Primed library functions which are compositions with match](#primed-library-functions-which-are-compositions-with-match)
+    - [Constructors and Selectors](#constructors-and-selectors)
+    - [Naming introduced lambda bound Term variables](#naming-introduced-lambda-bound-term-variables)
+    - [Existential quantifiers](#existential-quantifiers)
+    - [Conditionals](#conditionals)
+    - [Explanations](#explanations)
+    - [Operations to define and use Specifications](#operations-to-define-and-use-specifications)
+    - [Utility functions](#utility-functions)
+    - [Escape Hatch to QuickCheck Gen monad](#escape-hatch-to-quickcheck-gen-monad)
+- [Strategy for constraining a large type with many nested sub-components.](#strategy-for-constraining-a-large-type-with-many-nested-sub-components)
+- [Writing HasSpec instances by hand.](#writing-hasspec-instances-by-hand)
+    - [Strategy 1 using GHC.Generics](#strategy-1-using-ghcgenerics)
+    - [Strategy 2 writing your own SimpleRep instance](#strategy-2-writing-your-own-simplerep-instance)
+    - [Strategy 3 defining the SimpleRep instance in terms of another type with a SimpleRep instance](#strategy-3-defining-the-simplerep-instance-in-terms-of-another-type-with-a-simplerep-instance)
+    - [Strategy 4, bypassing SimpleRep, and write the HasSpec instance by Hand](#strategy-4-bypassing-simplerep-and-write-the-hasspec-instance-by-hand)
+- [A look into the internals of the system.](#a-look-into-the-internals-of-the-system)
+
+<!-- markdown-toc end -->
+
+
 ## Constrained Generators is a First-Order Logic
 
 
 A First-order typed logic has 4 components
```

Comment:
I'm surprised to see that the description of a typed logic doesn't mention types at all.

Reply:
It does now. Thanks for the good idea.